app/celery/tasks.py

```diff
@@ -453,15 +451,18 @@ def save_emails(self, service_id: str, signed_notification: List[Any], receipt:
             notification["created_at"] = datetime.utcnow()
             notification["job_id"] = notification.get("job", None)
             notification["job_row_number"] = notification.get("row_number", None)
-            decrypted_notifications.append(notification)
+            verified_notifications.append(notification)
             notification_id_queue[notification_id] = notification.get("queue")
 
     try:
-        # this task is used by two main things... process_job and process_sms_or_email_notification
-        # if the data is not present in the encrypted data then fallback on whats needed for process_job
-        saved_notifications = persist_notifications(decrypted_notifications)
+        # If the data is not present in the encrypted data then fallback on whats needed for process_job
+        saved_notifications = persist_notifications(verified_notifications)
+        if receipt:
+            email_queue.acknowledge(receipt)
+            current_app.logger.info(f"Batch saving: {receipt} removed from buffer queue.")
     except SQLAlchemyError as e:
-        handle_list_of_exception(self, decrypted_notifications, e, receipt)
+        signed_and_verified = list(zip(signed_notifications, verified_notifications))
+        handle_batch_error_and_forward(self, signed_and_verified, EMAIL_TYPE, e, receipt)
```

Comment:
I like this pattern of sending the failed batch through to process each notification individually and then removing the list from the Redis queue.

Reply:
I thought we initially did this but I was wrong. It's a bit weird to have an idea but based on a mistake. Steve also proposed the same after the incident happened, so it seems quite natural.